packages/flutter/test_private/bin/test_private.dart

```diff
@@ -0,0 +1,208 @@
+final Directory privateTestsDir = Directory(path.join(testPrivateDir.path, 'test'));
+
+void _usage() {
+  print('Usage: test_private.dart [--help] [--temp-dir=<temp_dir>]');
+  print('''
+    --help      Print a usage message.
+    --temp-dir  A location where temporary files may be written. Defaults to a
+                directory in the system temp folder. If a temp_dir is not
+                specified, then the default temp_dir will be created, used, and
+                removed automatically.
+    ''');
+}
+
+Future<void> main(List<String> args) async {
+  // TODO(gspencergoog): Convert to using the args package once it has been
+  // converted to be non-nullable by default.
+  if (args.isNotEmpty && args[0] == '--help') {
+    _usage();
+    exit(0);
+  }
+
+  void errorExit(String message, {int exitCode = -1}) {
+    stderr.write('Error: $message\n\n');
+    _usage();
+    exit(exitCode);
+  }
+
+  if (args.length > 2) {
+    errorExit('Too many arguments.');
+  }
+
+  String? tempDirArg;
+  if (args.isNotEmpty) {
+    if (args[0].startsWith('--temp-dir')) {
+      if (args[0].startsWith('--temp-dir=')) {
+        tempDirArg = args[0].replaceFirst('--temp-dir=', '');
+      } else {
+        if (args.length < 2) {
+          errorExit('Not enough arguments to --temp-dir');
+        }
+        tempDirArg = args[1];
+      }
+    } else {
+      errorExit('Invalid arguments ${args.join(' ')}.');
+    }
+  }
+
+  Directory tempDir;
+  bool removeTempDir = false;
+  if (tempDirArg == null || tempDirArg.isEmpty) {
+    tempDir = Directory.systemTemp.createTempSync('flutter_package.');
+    removeTempDir = true;
+  } else {
+    tempDir = Directory(tempDirArg);
+    if (!tempDir.existsSync()) {
+      errorExit("Temporary directory $tempDirArg doesn't exist.");
+    }
+  }
+
+  try {
+    await for (final TestCase testCase in getTestCases(tempDir)) {
+      stderr.writeln('Running test case $testCase');
+      if (!await testCase.runTests()) {
+        stderr.writeln('Test case $testCase failed.');
+        exit(1);
+      } else {
+        stderr.writeln('Test case $testCase succeeded.');
+      }
+    }
+  } finally {
+    if (removeTempDir) {
+      tempDir.deleteSync(recursive: true);
+    }
+  }
+  exit(0);
+}
+
+File makeAbsolute(File file, {Directory? workingDirectory}) {
+  workingDirectory ??= Directory.current;
+  return File(path.join(workingDirectory.absolute.path, file.path));
+}
+
+/// A test case representing a private test file that should be run.
+///
+/// It is loaded from a JSON manifest file that contains a list of dependencies
+/// to copy, a list of test files themselves, and a pubspec file.
+///
+/// The dependencies are copied into the test area with the same relative path.
+///
+/// The test files are copied to the root of the test area.
+///
+/// The pubspec file is copied to the root of the test area too, but renamed to
+/// "pubspec.yaml".
+class TestCase extends Object {
```

Comment:
"extends Object" is redundant

Reply:
Yeah, I know, but there were some weird analysis problems with non-nullable code that it was fixing.  It appears to not be needed now though, so maybe it was a bug in the analyzer that got fixed.